src/client.rs

```diff
@@ -188,6 +189,9 @@ pub struct ClientConfig {
     /// Supported versions, in no particular order.  The default
     /// is all supported versions.
     pub versions: Vec<ProtocolVersion>,
+
+    /// How to verify the server certificate chain.
+    verifier: &'static verify::ServerCertVerifier,
```

Comment:
`Box` also works, the ability to have some state on the verifier can be important for some use cases :-)